scripts/esm/grid-renderer.mjs

```diff
@@ -309,14 +309,11 @@ class GridRenderer extends Script {
 
         const targetLayer = layer ?? GridRenderer.createLayer(this.app, undefined, 1);
 
-        const camera = this.entity.camera;
-        if (!camera) {
-            throw new Error('GridRenderer script must be attached to a camera entity.');
-        }
+        this.app.scene.on('prerender:layer', (camera, layer, transparent) => {
+            if (!camera.layers.includes(targetLayer.id)) {
+                camera.layers = camera.layers.concat(targetLayer.id);
```

Comment:
I really don't like changing camera.layers from inside the rendering callback.
That means at least one frame will be rendered incorrectly, and could even cause some crashes, as for example meshes on that layer were not culled for the camera, but in the middle of the frame, camera new needs to render them.

Reply:
Ahh i see what would you suggest I change it to instead?

Reply:
So ive currently changed it to store the cameras in the prerender but update the layers in an update loop. Would that solve the issue?

Reply:
I guess theres no other way then and the grid has to be attached to the camera

Reply:
Or the grid could be assigned a layer, that's what it really needs right? So that the mesh instance is added to that layer.
Inside the preRender callback, a uniform with the camera properties can be changed on the material, I believe there was this need as well.

Reply:
So the grid does have a layer set so I could add a hook on there but does that pass the camera as parameter?

Not sure what you mean about the material tho

Reply:
this one does:
https://github.com/playcanvas/engine/blob/c26ecf6ac9195b559dfa9f03df8f1ceda7be0e61/src/scene/scene.js#L109-L120

Reply:
I thought you needed to set some camera view matrix or similar uniform, not entirely sure. Ignore if not.

Reply:
Not set I just need to get the projection and the frustum corners